www/pfSense-pkg-squid/files/usr/local/www/status_squid.php

```diff
@@ -0,0 +1,110 @@
+$shortcut_section = "squid";
+include("head.inc");
+
+$tab_array = array();
+if ($_REQUEST["menu"] == "reverse") {
+	$tab_array[] = array(gettext("General"), false, "/pkg_edit.php?xml=squid_reverse_general.xml&amp;id=0");
+	$tab_array[] = array(gettext("Web Servers"), false, "/pkg.php?xml=squid_reverse_peer.xml");
+	$tab_array[] = array(gettext("Mappings"), false, "/pkg.php?xml=squid_reverse_uri.xml");
+	$tab_array[] = array(gettext("Redirects"), false, "/pkg.php?xml=squid_reverse_redir.xml");
+	$tab_array[] = array(gettext("Real Time"), true, "/squid_monitor.php?menu=reverse");
+	$tab_array[] = array(gettext("Sync"), false, "/pkg_edit.php?xml=squid_reverse_sync.xml");
+} else {
+	$tab_array[] = array(gettext("General"), false, "/pkg_edit.php?xml=squid.xml&amp;id=0");
+	$tab_array[] = array(gettext("Remote Cache"), false, "/pkg.php?xml=squid_upstream.xml");
+	$tab_array[] = array(gettext("Local Cache"), false, "/pkg_edit.php?xml=squid_cache.xml&amp;id=0");
+	$tab_array[] = array(gettext("Antivirus"), false, "/pkg_edit.php?xml=squid_antivirus.xml&amp;id=0");
+	$tab_array[] = array(gettext("ACLs"), false, "/pkg_edit.php?xml=squid_nac.xml&amp;id=0");
+	$tab_array[] = array(gettext("Traffic Mgmt"), false, "/pkg_edit.php?xml=squid_traffic.xml&amp;id=0");
+	$tab_array[] = array(gettext("Authentication"), false, "/pkg_edit.php?xml=squid_auth.xml&amp;id=0");
+	$tab_array[] = array(gettext("Users"), false, "/pkg.php?xml=squid_users.xml");
+	$tab_array[] = array(gettext("Real Time"), false, "/squid_monitor.php");
+	$tab_array[] = array(gettext("Status"), true, "/status_squid.php");
+	$tab_array[] = array(gettext("Sync"), false, "/pkg_edit.php?xml=squid_sync.xml");
+}
+display_top_tabs($tab_array);
+
+function squid_status() {
+	global $config;
+
+	if (is_service_running('squid')) {
+		init_config_arr(array('installedpackages', 'squidcache','config'));
+		$proxy_ifaces = explode(",", $config['installedpackages']['squid']['config'][0]['active_interface']);
+		foreach ($proxy_ifaces as $iface) {
+			if (get_interface_ip($iface)) {
+				$ip = get_interface_ip($iface);
+				$lip = '127.0.0.1';
+			} else {
+				$ip = get_interface_ipv6($iface);
+				$lip = '::1';
+			}
+			exec("/usr/local/sbin/squidclient -l {$lip} -h {$ip} mgr:info", $result);
+		}
+	} else {
+		return(gettext('Squid Proxy is not running.'));
+	}
+	$i = 0;
+	$matchbegin = "Squid Object Cache";
+	foreach ($result as $line) {
+		if (preg_match("/{$matchbegin}/", $line)) {
+			$begin = $i;
+		}
+		$i++;
+	}
+	
+	$output = "";
+	$i = 0;
+	
+	foreach ($result as $line) {
+		if ($i >= $begin) {
+			$output .= $line . "\n";
+		}
+		$i++;
+	}
+	return $output;
+}	
+
+?>
+
+<div class="panel panel-default">
+        <div class="panel-heading"><h2 class="panel-title">Connection list</h2></div>
+        <div class="panel-body table-responsive">
+        <table class="table table-striped table-hover table-condensed">
+        <tbody>
+		<?php 
+		print "<pre>";
+		print squid_status();
```

Comment:
Should run that through `htmlentities()` before printing as well, for safety.

Reply:
Fixed